examples/stopwatch/client.js

```diff
@@ -0,0 +1,93 @@
+const choo = require('../../')
+const html = require('../../html')
+const xtend = require('xtend')
+const raf = require('raf')
+
+const app = choo()
+
+function format (elapsed) {
+  const minutes = formatMinutes(Math.floor((elapsed / 1000) / 60))
+  const seconds = formatSeconds(Math.floor((elapsed / 1000) % 60))
+  const ms = formatSeconds(Math.floor((elapsed % 1000) / 10))
+
+  return `${minutes}:${seconds}.${ms}`
+}
+
+function formatMinutes (minutes) {
+  return `${minutes >= 10 ? minutes : (minutes < 10) ? '0' + minutes : '00'}`
+}
+
+function formatSeconds (seconds) {
+  return `${seconds < 10 ? '0' + seconds : seconds}`
+}
+
+function reset (send) {
+  send('laps:clear')
+  send('stopwatch:update', 0)
+}
+
+function lap (send, elapsed) {
+  send('laps:add', format(elapsed))
+}
+
+function toggle (send) {
+  send('stopwatch:toggle')
+}
+
+app.model({
+  namespace: 'stopwatch',
+  state: {
+    elapsed: 0,
+    start: false
+  },
+  reducers: {
+    toggle: (data, state) => xtend(state, { start: !state.start }),
+    update: (data, state) => xtend(state, { elapsed: data })
+  }
+})
+
+app.model({
+  namespace: 'laps',
+  state: {
+    laps: []
+  },
+  reducers: {
+    add: (data, state) => xtend(state, { laps: state.laps.concat(data) }),
+    clear: (data, state) => ({ laps: [] })
+  }
+})
+
+const mainView = (state, prev, send) => {
+  const elapsed = state.stopwatch.elapsed
+  const now = Date.now() - elapsed
+  const start = state.stopwatch.start
+
+  function step () {
+    const elapsed = Math.floor((Date.now() - now))
+    send('stopwatch:update', elapsed)
+  }
+
+  if (start) {
```

Comment:
Hmm, I think that this piece would be better off in a `subscription`; not as part of a `view` - it _is_ a read-only source of data after all; similar to the `elm` time example http://elm-lang.org/examples/time